packages/llm/distributed.py

```
@@ -0,0 +1,338 @@
+tokens = math.min(capacity, tokens + elapsed * rate)
+
+local allowed = 0
+if tokens >= requested then
+  tokens = tokens - requested
+  allowed = 1
+end
+
+redis.call('HMSET', key, 'tokens', tokens, 'updated', now)
+redis.call('EXPIRE', key, 3600)
+return {allowed, tostring(tokens)}
+"""
```

Ruff (0.16.0)

[error] Possible hardcoded password assigned to: "_TOKEN_BUCKET_LUA"

(S105)